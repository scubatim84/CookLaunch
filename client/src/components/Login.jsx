import React from 'react';
import { CardMedia, Grid } from '@material-ui/core';

import LoginForm from './Auth/LoginForm';
import useStylesMain from '../Styles';

const Login = (props) => {
  const classes = useStylesMain();

  return (
    <CardMedia className={classes.loginBackground}>
      <Grid
        container
        alignItems="center"
        justify="center"
        style={{ minHeight: '100vh' }}
      >
        <Grid item xs={12} sm={6}>
          <LoginForm
            key={props.isLoggedIn}
            handleLoggedIn={props.handleLoggedIn}
            isLoggedIn={props.isLoggedIn}
          />
        </Grid>
      </Grid>
    </CardMedia>
  );
};

export default Login;
