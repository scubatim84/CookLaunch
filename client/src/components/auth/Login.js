import React, { useState } from "react";
import { Link } from "react-router-dom";

function Login() {
	const [registerUser, setRegisterUser] = useState({
		email: "",
		password: "",
	});
	const [error, setError] = useState({
		errorMessage: ""
	});

	const onChange = e => {
    setRegisterUser({ [e.target.id]: e.target.value });
  };

  const onSubmit = e => {
    e.preventDefault();
    
    const userData = {
      email: registerUser.email,
      password: registerUser.password
    };
	};
	
	return (
		<div className="container">
			<div style={{ marginTop: "4rem" }} className="row">
				<div className="col s8 offset-s2">
					<Link to="/" className="btn-flat waves-effect">
						<i className="material-icons left">keyboard_backspace</i> Back to
						home
					</Link>
					<div className="col s12" style={{ paddingLeft: "11.250px" }}>
						<h4>
							<b>Login</b> below
						</h4>
						<p className="grey-text text-darken-1">
							Don't have an account? <Link to="/register">Register</Link>
						</p>
					</div>
					<form noValidate onSubmit={this.onSubmit}>
						<div className="input-field col s12">
							<input
								onChange={this.onChange}
								value={this.state.email}
								id="email"
								name="email"
								type="email"
							/>
							<label htmlFor="email">Email</label>
						</div>
						<div className="input-field col s12">
							<input
								onChange={this.onChange}
								value={this.state.password}
								id="password"
								name="password"
								type="password"
							/>
							<label htmlFor="password">Password</label>
						</div>
						<div className="col s12" style={{ paddingLeft: "11.250px" }}>
							<button
								style={{
									width: "150px",
									borderRadius: "3px",
									letterSpacing: "1.5px",
									marginTop: "1rem"
								}}
								type="submit"
								className="btn btn-large waves-effect waves-light hoverable blue accent-3"
							>
								Login
							</button>
						</div>
					</form>
				</div>
			</div>
		</div>
	);
}

export default Login;