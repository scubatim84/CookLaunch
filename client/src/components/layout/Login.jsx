import React from 'react';
import LoginForm from '../auth/LoginForm';

import Grid from '@material-ui/core/Grid';
import {makeStyles} from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
}));

function Login(props) {
  const classes = useStyles();

  return (
    <div className='login-background background-image-full'>
      <div className={classes.root}>
        <Grid
          container
          spacing={0}
          alignItems='center'
          justify='center'
          style={{minHeight: '100vh'}}
        >
          <Grid className={classes.position} item xs={6}>
            <LoginForm
              handleLoggedIn={props.handleLoggedIn}
              isLoggedIn={props.isLoggedIn}
            />
          </Grid>
        </Grid>
      </div>
    </div>
  );
}

export default Login;
