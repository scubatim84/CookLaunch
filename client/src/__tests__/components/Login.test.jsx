import React from 'react';
import { render } from '@testing-library/react';

import Login from '../../components/Login';

describe('Login', () => {
  it('Renders component without crashing', () => {
    render(<Login />);
  });
});
