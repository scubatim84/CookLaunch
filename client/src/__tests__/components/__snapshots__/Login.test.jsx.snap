import React from 'react';
import {shallow} from 'enzyme';
import Login from '../../components/Login';

describe('Login', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(<Login />);

    expect(wrapper).toMatchSnapshot();
  });
});
